server/infrastructure/saved_players_repository.py

```
@@ -0,0 +1,111 @@
+                detail="Firebase is not configured"
+            )
+        
+        try:
+            player_ref = self.db.collection('users').document(user_id).collection('saved_players').document(player_id)
+            player_doc = player_ref.get()
+            
+            if not player_doc.exists:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    detail=f"Player with ID {player_id} not found"
+                )
```

Ruff (0.14.1)

Abstract `raise` to an inner function

(TRY301)